core/io/resource.cpp

```diff
@@ -514,6 +514,17 @@ Resource *ResourceCache::get(const String &p_path) {
 	return *res;
 }
 
+void ResourceCache::forget(const String &p_path) {
+	lock.write_lock();
+	if (!resources.has(p_path)) {
+		lock.write_unlock();
+		ERR_FAIL_MSG("Cannot forget a cached resource " + p_path + " as it was not present in the cache!");
+		return;
+	}
+	resources.erase(p_path);
+	lock.write_unlock();
+}
```

Comment:
I wouldn't add it if it's not needed.

We don't need to have feature parity here between `3.x` and `4.0` as 4.0 already has a significantly different cache system.

Reply:
In that case, I've removed it.